### PR TITLE
openhmd-0.1.0

### PR DESCRIPTION
**Pull request by PyrApple**
Added Formula for openhmd-0.1.0.
(my first contribution to homebrew , may have overlooked some details)

### CODE REVIEW

File: Library/Formula/openhmd.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  homepage "http://openhmd.net"
+
+  url "http://openhmd.net/releases/openhmd-0.1.0.tar.gz"
+  sha1 "186c747399bd9a509ac8300acbae8823fc4fcc79"
+
+  head "https://github.com/OpenHMD/OpenHMD.git", :branch => "master"
+
+  head do
+    sha1 "bd475d6a2992d0729e3c376a013ff97f3bbce387"
+  end
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Just remove this, it'll use the `master` branch above.

File: Library/Formula/openhmd.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "autoconf" => :build # if built from git
+  depends_on "automake" => :build # if built from git
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: You can remove these comments above.

Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Also can probably all be in a `head do` block if that's the case.

File: Library/Formula/openhmd.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+class Openhmd < Formula
+  homepage "http://openhmd.net"
+
+  url "http://openhmd.net/releases/openhmd-0.1.0.tar.gz"
+  sha1 "186c747399bd9a509ac8300acbae8823fc4fcc79"
+
+  head "https://github.com/OpenHMD/OpenHMD.git", :branch => "master"
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Move this to an `url` line in `head do` below.

File: Library/Formula/openhmd.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  depends_on "automake" => :build # if built from git
+  depends_on "pkg-config" => :build # not sure yet
+  depends_on "hidapi"
+  depends_on "libtool" => :build if build.head?
+
+  def install
+
+    if not build.head?
+      opoo "run `brew install openhmd --HEAD' (from git HEAD) for DK2 support (02/2015)"
+    end
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Remove this.

File: Library/Formula/openhmd.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+    args = ["--prefix", prefix,
+            "--disable-debug",
+            "--disable-silent-rules",
+            "--disable-dependency-tracking"]
+
+    system "./autogen.sh" if build.head? # (if building openhmd from the git repository)
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Remove the comment here.

File: Library/Formula/openhmd.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+    system "./configure", *args
+
+    system "make", "install"
+  end
+
+  test do
+    system "false"
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

**Comment by PyrApple** [comment]
pending checks done according to mikemcquaid  propositions.

File: Library/Formula/openhmd.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
+class Openhmd < Formula
+  homepage "http://openhmd.net"
+
```
Comment (DomT4) [inline Library/Formula/openhmd.rb]: Chop the whitespace between ` homepage ` and ` url ` :)

File: Library/Formula/openhmd.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  url "http://openhmd.net/releases/openhmd-0.1.0.tar.gz"
+  sha1 "186c747399bd9a509ac8300acbae8823fc4fcc79"
+
+  head do
+    url "https://github.com/OpenHMD/OpenHMD.git", :branch => "master"
```
Comment (DomT4) [inline Library/Formula/openhmd.rb]: ` master ` is the default branch and doesn't usually need stating.

File: Library/Formula/openhmd.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  depends_on "hidapi" => :build
+
+  def install
+    args = ["--prefix", prefix,
+            "--disable-debug",
+            "--disable-silent-rules",
+            "--disable-dependency-tracking"]
+
+    system "./autogen.sh" if build.head?
+
```
Comment (DomT4) [inline Library/Formula/openhmd.rb]: Probably good to kill whitespace down here too.

```ruby
system "./autogen.sh" if build.head?
system "./configure", *args
system "make", "install"
```

Is usually fine.

File: Library/Formula/openhmd.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    system "./configure", *args
+
+    system "make", "install"
+  end
+
+  test do
+    system "unittests"
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: This shouldn't be installed into the `bin` directory; the name is  too generic. Also: `system "#{bin}/unittests"`

**Comment by PyrApple** [comment]
removed white space following DomT4 comments, 
did not yet updated following mikemcquaid request:
shouldn't be installed into the bin directory; the name is too generic. Also: system "#{bin}/unittests"

**Comment by PyrApple** [comment]
following mikemcquaid request, installed binary into #{prefix}/bin/ rather than #{bin}

File: Library/Formula/openhmd.rb [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "hidapi" => :build
+
+  def install
+    args = ["--prefix", prefix,
+            "--disable-debug",
+            "--disable-silent-rules",
+            "--disable-dependency-tracking",
+            "--bindir=#{prefix}/bin"]
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: `#{prefix}/bin` is the same as `#{bin}`

File: Library/Formula/openhmd.rb [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+    system "./autogen.sh" if build.head?
+    system "./configure", *args
+    system "make", "install"
+  end
+
+  test do
+    system "#{prefix}/bin/unittests"
```
Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Sorry, I meant just the `unittests` library specifically is named too generically to go into `bin`. As the last line of `openhmd` about you could do `(share+"tests").install "#{bin}/unittests"` which will move it after installation.

Comment (PyrApple) [inline Library/Formula/openhmd.rb]: Modified according to Mike's suggestions.

apologies for the clumsiness :)

**Comment by PyrApple** [comment]
Modified according to Mike's suggestions.
apologies for the clumsiness :)

File: Library/Formula/openhmd.rb (hunk 12 above)
Comment (PyrApple) [inline Library/Formula/openhmd.rb]: I was considering the addition of the ln -s of 

Cellar/openhmd/HEAD/include/openhmd/openhmd.h
to 
Cellar/python3/3.4.1_1/Frameworks/Python.framework/Versions/3.4/include/python3.4m/

to build python-rift developed @ https://github.com/lubosz/python-rift

Do you think it's worth adding it to the formula as an option or is it too much specific?

David

Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: I don't think we want this formula to be writing into the Python Cellar so pass on that for now.

Comment (PyrApple) [inline Library/Formula/openhmd.rb]: Fine by me, is the openhmd formula sound as is?

Comment (mikemcquaid) [inline Library/Formula/openhmd.rb]: Yup, looks good, will merge shortly!

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `openhmd 0.1.0 (new formula)`.

**Pull request closed without merging**